### DISCUSSION TITLE
RLS Policy on Insert

### DISCUSSION THREAD

**Discussion by inbytesdevteam** [supabase/supabase#270] [Questions] [answered]
Hi! I read up on [Row Level Security](https://www.postgresql.org/docs/current/ddl-rowsecurity.html) and I'm trying to implement it in our supabase database.

We have a table of `subscribers` which should allow `SELECT` access to admin users, and `INSERT` access to all, even without logging in.

I've already implemented the read access part for admin users, but I'm having trouble with the `INSERT` part. Here's the policy I added:
```sql
CREATE POLICY "Allow all insert access" on subscribers FOR INSERT WITH CHECK (true);
```
![image](https://user-images.githubusercontent.com/70461143/97769315-beff1600-1b64-11eb-9559-663561d372fe.png)

However, when I try to insert a subscriber, this error shows up:
```js
Uncaught (in promise) Error: {"hint":null,"details":null,"code":"42501","message":"new row violates row-level security policy for table \"subscribers\""}
```

Hope I can ask for help on this! I also checked if the docs already have support for Policies and saw that it is still in the list of issues to do.

**Comment by steve-chavez** [comment]
Hey @inbytesdevteam 

I think this could be an issue with postgrest-js(which supabase-js uses underlyingly). I've reported it here: https://github.com/supabase/postgrest-js/issues/127.

**Comment by soedirgo** [comment, marked as answer]
Hey @inbytesdevteam! Sorry this took so long.

This is now merged in supabase-js, and will be released in the next version. Then you can do:
```js
await supabase.from('subscribers')
    .insert({...}, { returning: 'minimal' })
```
And likewise for other write-only tables. Ref: https://github.com/supabase/postgrest-js/pull/133

**Reply by inbytesdevteam** [reply]
Thank you @soedirgo and team. Can confirm that this works!

It should also probably be added to the docs as well, related to the [Auth Policies](https://supabase.io/docs/guides/auth#policies) perhaps?

**Reply by soedirgo** [reply]
Awesome, thanks! Definitely good to include it in the Tips, and in the client docs as well.

**Reply by LeoHeuser** [reply]
Exactly what I need. @soedirgo, do you know how to implement this in Swift?

Thanks and best ✌🏼

**Reply by soedirgo** [reply]
I'm not familiar with postgrest-swift, but this just adds an HTTP header (`Prefer: return=minimal`) so shouldn't be difficult to implement.

**Reply by LeoHeuser** [reply]
Thx for the tip. You were right. You can pass it in an array with the client in Swift.

Sometimes but you have to write "prefer" with lower letters. And sometimes it still shows a security violation - seems a bit buggy.

But this helps me a lot. Thx again. ✌🏼

**Reply by siisee11** [reply]
There is no options named returning on insert function.

**Reply by steve-chavez** [reply]
@siisee11 Nowadays you do `insert().select()` to get the result from an insert. `select()` is not done by default for `insert()` or other mutations.
